### PR TITLE
CHE-6733: Implement User Preferences for jdt.ls

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Implements User Preferences for jdt.ls

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6733

### CODE REVIEW

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 1]
```diff
@@ -633,6 +651,36 @@ public WorkspaceEdit organizeImports(String path) {
     return doGetOne("java.edit.organizeImports", singletonList(prefixURI(path)), type);
   }
 
+  // configuration
```
Comment (svor) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java:699]: Do we really need this comment?

**Review by svor — APPROVED** [review]

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/service/JavaLanguageExtensionServiceClient.java [hunk 2]
```diff
@@ -295,6 +298,40 @@ private void subscribe() {
...
+                .endpointId(WS_AGENT_JSON_RPC_ENDPOINT_ID)
+                .methodName(GET_JAVA_CORE_OPTIONS)
+                .paramsAsListOfString(filter)
+                .sendAndReceiveResultAsDto(JavaCoreOptions.class, REQUEST_TIMEOUT)
+                .onSuccess(resolve::apply)
+                .onTimeout(() -> onTimeout(reject))
+                .onFailure(error -> reject.apply(ServiceUtil.getPromiseError(error))));
+  }
+
+  public Promise<Boolean> updateJavaCoreOptions(JavaCoreOptions javaCoreOptions) {
```
Comment (mmorhun) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/service/JavaLanguageExtensionServiceClient.java:327]: Please add a comment how it behaves (merge or replace existing settings. What happened if an empty map given, etc.)

**Review by mmorhun — APPROVED** [review]

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 3]
```diff
@@ -633,6 +651,36 @@ public WorkspaceEdit organizeImports(String path) {
...
+  /** Updates JDT LS java core options. */
+  public Boolean updateJavaCoreOptions(JavaCoreOptions javaCoreOptions) {
+    Type type = new TypeToken<Boolean>() {}.getType();
+    return doGetOne(
+        Commands.UPDATE_JAVA_CORE_OPTIONS_СOMMAND, singletonList(javaCoreOptions), type);
+  }
+  /** Updates JDT LS preferences. */
+  public Boolean updatePreferences(JdtLsPreferences preferences) {
+    Type type = new TypeToken<Boolean>() {}.getType();
+    return doGetOne(Commands.UPDATE_PREFERENCES_СOMMAND, singletonList(preferences), type);
```
Comment (tsmaeder) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: Is "update preferences" called anywhere? If not, let's not add commands that we don't use.

**Review by skabashnyuk — APPROVED** [review]

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java (hunk 3 above)
Comment (tolusha) [reply plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: ok

**Pull request merged**